### PR TITLE
clean some SparseArrays from base docs

### PR DESCRIPTION
**Pull request by fredrikekre**
Remove some sparse related stuff from base docs, with the aim to decouple even more and make base doctest work without `SparseArrays`.

### CODE REVIEW

**Review by ararslan — APPROVED** [review]

File: base/abstractarray.jl [hunk 1]
```diff
@@ -801,39 +801,29 @@ range to efficiently index into the array with indices specified for every dimen
...
 If you supply more than one `AbstractArray` argument, `eachindex` will create an
 iterable object that is fast for all arguments (a `UnitRange`
 if all inputs have fast linear indexing, a [`CartesianIndices`](@ref)
 otherwise).
 If the arrays have different sizes and/or dimensionalities, `eachindex` will return an
 iterable that spans the largest range along each dimension.
+
+# Examples
+```jldoctest
+julia> for i in eachindex(A) # linear indexing
```
Comment (Sacha0) [inline base/abstractarray.jl]: Am I missing the definition of `A` somewhere above? :)

File: base/essentials.jl [hunk 2]
```diff
@@ -106,37 +106,16 @@ julia> convert(Rational{Int64}, x)
...
-
-julia> y === x
-true
-
-julia> z === x
-false
-
-julia> z.colptr === x.colptr
-true
-```
```
Comment (Sacha0) [inline base/essentials.jl]: Preserving an equivalently incisive illustration of this caveat might be nice if possible?

File: doc/src/devdocs/subarrays.md [hunk 3]
```diff
@@ -17,7 +17,7 @@ allows you to combine these styles of indexing: for example, a 3d array `A3` can
 For `Array`s, linear indexing appeals to the underlying storage format: an array is laid out as
 a contiguous block of memory, and hence the linear index is just the offset (+1) of the corresponding
 entry relative to the beginning of the array.  However, this is not true for many other `AbstractArray`
-types: examples include [`SparseMatrixCSC`](@ref), arrays that require some kind of
+types: examples include sparse arrays, arrays that require some kind of
```
Comment (Sacha0) [inline doc/src/devdocs/subarrays.md]: Perhaps reference the stdlib module?

File: base/abstractarray.jl (hunk 1 above)
Comment (fredrikekre) [reply base/abstractarray.jl]: Woops, thanks.

File: base/essentials.jl (hunk 2 above)
Comment (fredrikekre) [reply base/essentials.jl]: Do you have a proposal? This example was very ugly... :slightly_smiling_face:

Comment (Sacha0) [reply base/essentials.jl]: Coming up with an appropriate illustration is left as an exercise to the reader 😉.

File: doc/src/manual/interfaces.md [hunk 4]
```diff
@@ -250,8 +250,8 @@ arrays are simple: just define `getindex(A::ArrayType, i::Int)`.  When the array
 indexed with a multidimensional set of indices, the fallback `getindex(A::AbstractArray, I...)()`
 efficiently converts the indices into one linear index and then calls the above method. `IndexCartesian()`
 arrays, on the other hand, require methods to be defined for each supported dimensionality with
-`ndims(A)` `Int` indices. For example, the built-in [`SparseMatrixCSC`](@ref) type only
-supports two dimensions, so it just defines
+`ndims(A)` `Int` indices. For example, [`SparseMatrixCSC`](@ref) from the `SparseArrays` standard
+library module, only supports two dimensions, so it just defines
```
Comment (Sacha0) [inline doc/src/manual/interfaces.md]: Errant comma after `module`? :)

**Review by Sacha0 — APPROVED** [review]
lgtm! Thanks @fredrikekre! :)

File: base/essentials.jl (hunk 2 above)
Comment (Sacha0) [reply base/essentials.jl]: Slack conversation recap: An equivalently incisive illustration is tricky to come up with, so scratch this comment :).

**Pull request merged**